framework/src/problems/ReferenceResidualProblem.C

```diff
@@ -337,25 +356,38 @@ ReferenceResidualProblem::updateReferenceResidual()
   System & s = nonlinear_sys.system();
   auto & as = aux_sys.sys();
 
-  for (unsigned int i = 0; i < _group_resid.size(); ++i)
-  {
-    _group_resid[i] = 0.0;
-    _group_output_resid[i] = 0.0;
-    _group_ref_resid[i] = 0.0;
-  }
+  std::fill(_group_resid.begin(), _group_resid.end(), 0.0);
+  std::fill(_group_output_resid.begin(), _group_output_resid.end(), 0.0);
+  if (_local_norm)
+    std::fill(_group_ref_resid.begin(), _group_ref_resid.end(), 1.0);
+  else
+    std::fill(_group_ref_resid.begin(), _group_ref_resid.end(), 0.0);
 
   for (unsigned int i = 0; i < _soln_vars.size(); ++i)
   {
-    const auto resid =
-        Utility::pow<2>(s.calculate_norm(nonlinear_sys.RHS(), _soln_vars[i], DISCRETE_L2));
```

Comment:
This is a bit weird that the norm is being squared (canceling the square root of the L2). It's not the L2 norm comparison, it's the sum of the squares?